core/src/main/java/org/javarosa/core/model/instance/TreeElement.java

```diff
@@ -1245,4 +1167,191 @@ public void addAttributeMap(Hashtable<XPathPathExpr, Hashtable<String, TreeEleme
     public Vector<TreeReference> tryBatchChildFetch(String name, int mult, Vector<XPathExpression> predicates, EvaluationContext evalContext) {
         return TreeUtilities.tryBatchChildFetch(this, mChildStepMapping, name, mult, predicates, evalContext);
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Implemented this for test purposes. I suspect it is pretty slow. I'm fairly sure it is never called in production code.